### ISSUE TITLE
"Hide whitespace changes" also hides changes inside strings

### ISSUE THREAD

**Issue by skjasm** [sublimehq/sublime_merge#1741]
**Version info**

 - OS: Ubuntu 20.04
 - Build: 2083

**Description**

"Hide whitespace changes" also hides whitespace changes inside strings

**Steps to reproduce**

1. Commit a file containing a string in quotes.
2. Edit file, add or remove whitespace in the string.
3. View the edited file in Sublime Merge with "hide whitespace changes enabled".

**Expected behavior**

I expected "Hide whitespace changes" to hide whitespace changes in code that do not affect the program: spaces around punctuation, indentation in C++. This works.

I expected Sublime Merge to still show all changes that affect the program. Changing the content of a string is always significant, even when the change is only whitespace in the string.

**Screenshots**

![a_demo_1](https://user-images.githubusercontent.com/113343278/227527392-9a13e8ec-f14c-4b7b-9218-3974d2c0b66a.png)
![a_demo_2](https://user-images.githubusercontent.com/113343278/227527398-3da2a63b-efb8-4c49-abb1-b6e358450980.png)

**Comment by jfcherng** [comment]
Nothing in Git is language context-aware, most likely. It's just text.

![image](https://user-images.githubusercontent.com/6594915/227530805-a5ee1ad6-c858-4b2e-8e72-08713c4d291f.png)

The fancy tool you want is probably https://github.com/Wilfred/difftastic

**Comment by skjasm** [comment]
Fair enough. SM is at some level aware of what's strings and what's not since it can color syntax, but I guess that information is not available to the diff engine.

**Comment by jfcherng** [comment]
I think SM doesn't do any diff by itself but just call the git executable. And then color the diff output from git with corresponding syntax definition.

**Comment by kfranqueiro** [comment]
Just stumbled upon this today and have to agree it makes the option kind of unusable if it's going to hide meaningful changes.

It seems like this must rely on `git diff -w` under the surface. Perhaps it should rely on `git diff -b` instead?
